src/dblib/bcp.c

```diff
@@ -2226,7 +2226,7 @@ _bcp_get_col_data(TDSBCPINFO *bcpinfo, TDSCOLUMN *bindcol, int offset TDS_UNUSED
 
 	/* if (Max) column length specified take that into consideration. */
 
-	if (bindcol->column_bindlen >= 0) {
+	/* if (bindcol->column_bindlen >= 0) */ { /* bindlen is unsigned */
```

Comment:
Not sure if this is what you want, or instead the variable should be signed instead. Depends on the usage, that variable is declared in libTDS but used in different ways by upper libraries. I need to check.

Reply:
OK, thanks.  It's been unsigned all along, but I suppose that doesn't mean it can't change.

Reply:
Just checked usage of this in db-library and in multiple cases the value can be -1 (everything less than -1 should be an error... which actually it's not properly checked :worried:). Actually if the type is fixed -1 is the value usually used!
So probably that change spot an error and code should be
```c
	if ((DBINT) bindcol->column_bindlen >= 0) {
```
wondering how many hidden bugs like this there could be in the code

Reply:
In that case, perhaps the field should be signed after all.  Warnings from supplying `-1` would have been helpful, but it's apparently deemed legitimate because `-1` is a legal *signed* `int`. :-/

Reply:
Merged everything but this hunk. Looking at all libraries it looks like all uses signed, I think it would be better to change the variable to signed.